enterprise/admin/__init__.py

```diff
@@ -135,12 +165,44 @@ class EnterpriseCustomerUserAdmin(admin.ModelAdmin):
     class Meta(object):
         model = EnterpriseCustomerUser
 
-    fields = ('user_id', 'enterprise_customer')
+    fields = (
+        'user_id',
+        'enterprise_customer',
+        'user_email',
+        'username',
+        'created',
+    )
+
+    # Only include fields that are not database-backed; DB-backed fields
+    # are dynamically set as read only.
+    readonly_fields = (
+        'user_email',
+        'username',
+        'created',
+    )
+
+    inlines = (
+        HistoricalUserDataSharingConsentAuditInlineAdmin,
+    )
+
+    def username(self, ecu):
+        """
+        Return the username for the attached user.
+        """
+        return ecu.user.username
 
     def get_readonly_fields(self, request, obj=None):
         """
         Make all fields readonly when editing existing model.
         """
         if obj:  # editing an existing object
-            return get_all_field_names(self.model)
-        return tuple()
+            return super(
+                EnterpriseCustomerUserAdmin,
+                self
+            ).get_readonly_fields(
+                request,
+                obj=obj
+            ) + tuple(
+                get_all_field_names(self.model)
+            )
+        return super(EnterpriseCustomerUserAdmin, self).get_readonly_fields(request, obj=obj)
```

Comment:
Nit: You have the code

    super(EnterpriseCustomerUserAdmin, self).get_readonly_fields(request, obj=obj)

twice, in both branches.  The code would become more readable when written like this:

    fields = super(EnterpriseCustomerUserAdmin, self).get_readonly_fields(request, obj=obj)
    if obj:
        fields += tuple(get_all_field_names(self.model))
    return fields